docs/globalnet.md

```diff
@@ -0,0 +1,71 @@
+# Introduction
+
+Submariner is a tool build to connect overlay networks of different Kubernetes clusters. These clusters can be on different public clouds or OnPremises. An important use case for Submariner is to connect disparage independent clusters into a single cohesive multi-cluster.
+ However, one key limitation of submariner's current design is that it doesn't support overlapping CIDRs across these clusters. Each cluster must use a distinct ClusterCIDR that doesn't conflict or overlap with any other cluster that is going to be part of multicluster.
+
+<!-- TODO: Add diagram showing the problem --> 
+
+This is largely problematic coz most actual deployments use the default CIDR for cluster, this end up using the same CIDR. Changing CIDR on existing clusters is a very disruptive process and requires a cluster restart. So we require a means to allow clusters with Overlapping CIDRs connect together with submariner.
+
+# Architecture
+
+To support overlapping CIDRs in clusters connected through submariner, we're introducing a new component called Global Private Network, or GlobalNet (`globalnet`). This GlobalNet is a virtual network with scope as submariner multi-cluster and CIDR 169.254.0.0/16. Each cluster is given a subnet from this Global Private Network, configured as new cluster parameter `GlobalCIDR`.
+
+Once configured, each service and pod that requires cross cluster access is allocated an IP, `GlobalIp` from this `GlobalCIDR` and it is annotated to the POD/Service object. This `GlobalIp` is used for all cross-cluster communication from and to this POD/Service. Routing and IPTable/OVS/OVN rules are configured to use this IP for Ingress/Egress. All address translations occur at Gateway node.
+
+This is achieved by a new controller, Submariner Globalnet.
+
+## submariner-globalnet
+
+Submariner GlobalNet provides cross cluster connectivity between pods and services using GlobalIp. Compiled as binary `submariner-globalnet`, it is the controller that is responsible for maintaining and allocating pool of globalIPs, annotating services and pods with their globalIp, and finally configuring the required rules on gateway node to provide cross cluster connectivity using GlobalIp. It mainly consists of two key components.
```

Comment:
nit: maintaining and "allocating pool of globalIPs" -> this might give an impression that globalnet allocates a bunch of globalIPs. Shall we modify the sentence to "maintaining the GlobalPool and allocating an IP from this pool to Pods/Services, annotating them accordingly.....

Feel free to ignore this comment if you think my interpretation is wrong.